nixos/modules/i18n/input-method/fcitx5.nix

```diff
@@ -5,7 +5,7 @@ with lib;
 let
   im = config.i18n.inputMethod;
   cfg = im.fcitx5;
-  addons = cfg.addons ++ optional cfg.enableRimeData pkgs.rime-data;
+  addons = cfg.addons;
   fcitx5Package = pkgs.fcitx5-with-addons.override { inherit addons; };
   whetherRimeDataDir = any (p: p.pname == "fcitx5-rime") cfg.addons;
```

Comment:
Is it better to move this condition into `config.i18n.inputMethod.rime.enable` and let fcitx5 and ibus add rime addons automatically?

Reply:
I create [a patch](https://gist.github.com/Vonfry/5cfe2343d72ca02a84a7a91749556ad3) for this suggestion **without test**.

Reply:
There are two changes in the patch.

1. This patch moves the `environment.variables, systemPackages, pathsToLink` part from the ibus/fcitx5 modules to the rime module, I think these configs should be defined in the ibus/fcitx5 modules because they are related to the details about how ibus/fcitx5-rime find rime data.
2. I think `config.i18n.inputMethod.rime.enable` is an acceptable change, but it has two disadvantages:
   * Compatibility issue: it's a breaking change to the user's configuration
   * The option will become misleading: users cannot enable rime alone, the option will only work when the proper input method supporting rime is enabled.

   Proper assertions can eliminate the second issue, the main concern is the compatibility issue.

Reply:
@Vonfry After reading how the `addons` option of fcitx5 and the `plugins` option of ibus works, how about moving `NIX_RIME_DATA_DIR` to the fcitx5 and ibus wrapper, use the path $out/share/rime-data ($out is the output path of the wrapper).

Reply:
> moving NIX_RIME_DATA_DIR to the fcitx5 and ibus wrapper

It is better than current method. Thanks.

> The option will become misleading: users cannot enable rime alone.

I never think this problem. You are right. Please forget my suggestion.